app/controllers/users.js

```diff
@@ -1,56 +1,36 @@
-const { validationResult } = require('express-validator/check');
 const bcrypt = require('bcrypt');
 const jwt = require('jsonwebtoken');
 const { User, Purchase } = require('../models');
 const axios = require('axios');
 const logger = require('../logger');
 const config = require('../../config').common.session;
-const { defaultError } = require('../errors');
+const { dbError, userUnauthorized, needsAdmin, albumNotPurchased } = require('../errors');
 
-function logDBError(res) {
-  return error => {
-    logger.error(`DB: ${error.errors[0]}`);
-    res.status(500).json({ error: error.errors[0].message });
-  };
-}
-
-function createUser(userParams, res) {
+function createUser(userParams, res, next) {
   userParams.password = bcrypt.hashSync(userParams.password, 10);
   User.create(userParams)
     .then(user => {
       logger.info(`User ${user.name} successfuly created!`);
       res.status(200).json(user);
     })
-    .catch(logDBError(res));
+    .catch(err => next(dbError(err)));
 }
 
 module.exports = {
   userCreate(req, res, next) {
-    const errors = validationResult(req);
-    if (!errors.isEmpty()) {
-      logger.error('User input invalid.');
-      logger.error(errors.array());
-      return res.status(422).json({ errors: errors.array() });
-    }
     const userRaw = req.body;
     userRaw.admin = false;
-    createUser(userRaw, res);
+    createUser(userRaw, res, next);
   },
 
   userNewSession(req, res, next) {
-    const errors = validationResult(req);
-    if (!errors.isEmpty()) {
-      logger.error('User input invalid.');
-      logger.error(errors.array());
-      return res.status(422).json({ errors: errors.array() });
-    }
     const { email, password } = req.body;
     User.scope('withPasswd')
       .findOne({ where: { email } })
       .then(user => {
         if (!user) {
           logger.error('Email is not registered.');
-          return res.status(401).json({ error: 'User auth failed. Check your email or password.' });
+          next(userUnauthorized);
```

Comment:
this won't end the execution so you should add a return

Reply:
fixed